symengine/univariate_int.h

```diff
@@ -0,0 +1,82 @@
+/**
+ *  \file polynomial_int.h
+ *  Class for Univariate Int Polynomial Base
+ **/
+#ifndef SYMENGINE_UNIVARIATE_INT_H
+#define SYMENGINE_UNIVARIATE_INT_H
+
+#include <symengine/basic.h>
+
+namespace SymEngine
+{
+
+template <typename Coeff, typename Container, typename Poly>
+class UnivariateIntBase : public Basic
+{
+protected:
+    unsigned int degree_;
+    RCP<const Symbol> var_;
+    Container container_;
```

Comment:
Probably, `flint::fmpq_polyxx` and `piranha::polynomial<integer_class, piranha::monomial<unsigned int>>`. I will make changes to the overall structure if something doesn't fit well.